pkg/prometheus/statefulset.go

```diff
@@ -871,6 +902,14 @@ func prefixedName(name string) string {
 	return fmt.Sprintf("prometheus-%s", name)
 }
 
+func prometheusName(name string, shard int32) string {
```

Comment:
Nit: Could we add this function closer to the one where it's used, or vice versa, we only seem to use it in expectedStatefulSetShardNames?

Reply:
Makes sense, will do. I'll also rename it to `prometheusNameByShard`, which makes it a bit clearer what it does.